integration/spark/src/main/scala/org/apache/spark/sql/CarbonEnv.scala

```diff
@@ -21,22 +21,24 @@ import org.apache.spark.scheduler.cluster.CoarseGrainedSchedulerBackend
 import org.apache.spark.sql.hive.CarbonMetastore
 
 import org.apache.carbondata.hadoop.readsupport.impl.RawDataReadSupport
-import org.apache.carbondata.spark.rdd.SparkCommonEnv
+import org.apache.carbondata.spark.rdd.SparkReadSupport
 
 case class CarbonEnv(carbonMetastore: CarbonMetastore)
 
 object CarbonEnv {
 
   @volatile private var carbonEnv: CarbonEnv = _
 
+  // set readsupport class global so that the executor can get it.
+  SparkReadSupport.readSupportClass = classOf[RawDataReadSupport]
```

Comment:
No, this is the key point, init is called in driver, see the comment, place here is to let executor get the read support class